### ISSUE TITLE
Onkyo HT-RC660 hangs after using Onkyo binding.

### ISSUE THREAD

**Issue by sercasyr** [openhab/openhab1-addons#2722]
Hi, I have a HT-RC660.

My issue is simple but very hard to debug:

After 1 week or so of using the Onkyo binding, the binding usually stops working.
In fact, I have to reset the device for the device to be able to accept connections again.

Once I reset it, the binding works again. If use the binding, the problem will eventually happen again. The more I use it, the sooner it happens.
It seems to me that the binding causes the device server to hang.

I am using the latest 1.7.0-RC OpenHab.
##

**Comment by teichsta** [comment]
@paulianttila any idea what could cause this behaviour?

**Comment by paulianttila** [comment]
@sercasyr Binding will jam or the receiver?

Pretty much impossible to do anything without debug logs. So you should enable trace level logging to Onkyo binding and try to catch the jam situation.

Add following line to lockback.xml

```
<logger name="org.openhab.binding.onkyo" level="TRACE" />
```
Reference to issue #1999

**Comment by sercasyr** [comment]
Hi,

both the binding and the device get hanged eventually.

Most of the time is the binding that stops working. 
In this case, I open Onkyo Remote 3, I connect to the device, I power it on (from stand-by to on) and off (from on to stand-by), and then the binding usually works again (no need to restart OpenHAB).

However, sometimes, it seems that the device is the one that gets hanged. 
In this case I usually reset it to factory settings. Sometimes, it is sufficient to unplug and re-plug the device from power. 

Right now I can't debug. I will when I am able and when I have a way to reproduce the error.

It seems to me that the issue happens because  when I don't plan to use the Onkyo device for a long time, I switch it off (I don't leave it in stand by, I switch it off completely from mains power). It looks like switching it off makes the connection status get crazy in either side,,,

The device is usually in stand by mode, because otherwise I wouldn't be able to connect to it. However I sometimes don't need to connect to it for a long time and I switch it off.

Is that something that I shouldn't do??
I do the same with a Samsung Smart TV and I have never had any problems connecting to it from OpenHAB for that reason...

**Comment by 9037568** [comment]
Really need a debug log of the failure to help resolve the issue. @sercasyr , can you provide one?

**Comment by sercasyr** [comment]
I think I have one, but I am not sure I captured the failure.

I spent last sunday (yesterday) testing OpenHAB in debug mode, and I wasn't
been able to make the Onklyo binding fail at all.

When I woke up today, the binding wasn't working, so it may be the failure
we are trying to debug.

I'll send the log file when I clean it. I have many bindings and there are
lots of unnecessary (possibly with private stuff) lines. I hope it can
bring some light, although I can't explain what I did to make it fail...





2016-12-18 0:38 GMT+01:00 Chris Carman <notifications@github.com>:
